### DISCUSSION TITLE
Intercepting, parallel routes and static build

### DISCUSSION THREAD

**Discussion by thomas-gldntp** [vercel/next.js#52254] [Help]
### Summary

Hello everyone!

I hope you're doing well. I am writing to discuss an issue I have encountered while using  the new features "intercepting routes" and "parallel routes" in Next.js 13, specifically within a Capacitor application. I have thoroughly reviewed the documentation, tested the "Nextgram" project, and watched videos to understand this new feature.

To provide some context, in order to implement this feature with Capacitor, I am using a static import that Capacitor reads from. As a result, I had to remove server-side rendering (SSR) since we do not have a server running, and I am unable to use React Server Components (RSC) as well. All my pages utilize the "useClient" directive, and I retrieve query parameters using the "useSearchParams" hook.

The problem arises when I run the project using the following steps:

1. Run `yarn build` (next build) to create a static build folder containing all the HTML/CSS/JS files for Capacitor to read.
2. Run `yarn cap sync`, which is a combination of `cap copy` and `cap update`.
3. Run `yarn dev` (next dev --port 3000) to launch a development server on localhost:3000.

In this scenario, the intercepting and parallel routes work to some extent. Occasionally, the modal appears, but other times the page is displayed instead.

However, when I run the project using the following steps:

1. Run `yarn build` (next build) to create a static build folder containing all the HTML/CSS/JS files for Capacitor to read.
2. Run `npx serve@latest out` to launch a server that reads the content of the "out" folder.

In this case, the intercepting and parallel routes do not work at all. The intercepting is not working and then we are redirect to the page instead of seeing the page content inside the modal.

My app folder structure is as follows:

```
app
├── favicon.ico
├── globals.css
├── home
│   └── dashboard
│       ├── @programsModal
│       │   ├── (...)programs
│       │   │   ├── logic.ts
│       │   │   ├── page.tsx
│       │   │   └── type.ts
│       │   └── default.ts
│       ├── default.ts
│       ├── layout.tsx
│       └── page.tsx
├── page.module.css
├── page.tsx
├── programs
│   ├── default.ts
│   └── page.tsx
```

Now, my question is: Are the modal, intercepting, and parallel routes supposed to work in this kind of situation?

Thank you for your attention, and I am eagerly looking forward to your responses and opinions regarding this matter! :smile:

### Additional information

```js
Dependencies I'm using : 
* Capacitor: "^5.0.0",
* @chakra-ui/react: "^2.6.1", 
* react: "18.2.0",
* next: "13.4.1"

next info command input : 
 Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.5.0: Mon Apr 24 20:51:50 PDT 2023; root:xnu-8796.121.2~5/RELEASE_X86_64
    Binaries:
      Node: 18.16.0
      npm: 9.5.1
      Yarn: 1.18.0
      pnpm: N/A
    Relevant packages:
      next: 13.4.1
      eslint-config-next: 12.3.4
      react: 18.2.0
      react-dom: 18.2.0
```

**Comment by cutsoy** [comment]
I ran into the same issue as well, but I did manage to find a reasonable workaround.

Let's assume that you have a `path` that's both a parallel and intercepting route. The static output will now look like this:

```
○ out
  ├ (.)path
    ├ index.html
    └ index.txt
  └ path
    ├ index.html
    └ index.txt
```

The problem is, none of the `(.)path` files are actually used after generating the static output.
In order to get it working, I copied all of the `(.)*/index.txt` files, and moved them to their non-`(.)` counterparts.

This will load the regular `.html` file on refresh, and load the modal `.txt` file on client-side navigation.

---

You can optionally automate the process with the script below:

```
{
    "scripts": {
        "fix": "find out -path \"*/(.)*\" -name \"*.txt\" -exec bash -c 'cp \"$0\" \"${0/(.)/}\"' \\{\\} \\;",
    }
}
```

Hope it works for you too!
